specs/commandline/sign.md

```diff
@@ -37,8 +37,44 @@ Flags:
       --plugin-config strings    {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values
       --signature-format string  signature envelope format, options: 'jws', 'cose' (default "jws")
   -u, --username string          username for registry operations (default to $NOTATION_USERNAME if not specified)
+      --use-image-manifest       optional for using OCI image manifest to store signatures in a registry
 ```
 
+## Use OCI image manifest to store signatures
+
+By default, Notation uses [OCI artifact manifest][oci-image-manifest] to store signatures in the registries. For backward compatibility, Notation supports using `OCI image manifest` to store the signature in the registries that implement partial or older versions of the OCI Image specification. Use flag `--use-image-manifest` to force Notation store the signatures using OCI image manifest. Note that Notation doesn't automatically switch to `OCI image manifest` for a failure of using `OCI artifact manifest`.
+
+Registries MAY not implement or enable the `Referrers API`, which is used by the client to fetch the referrers. In the context of Notation, the referrers are signatures. Notation follows the fallback procedure defined in [OCI distribution spec][oci-backward-compatibility] if `Referrers API` is unavailable.
+
+### Set config property for OCI image manifest
+
+OCI image manifest requires additional property `config` of type `descriptor`, which is not required by OCI artifact manifest. Notation creates a default config descriptor for the user if the flag `--use-image-manifest` is used.
+
+Notation uses empty JSON object `{}` as the default configuration content, and thus the default `config` property is fixed, as following:
+
+```json
+"config": {
+    "mediaType": "application/vnd.cncf.notary.signature",
+    "size": 2,
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a"
```

Comment:
why size of 2? Also, can we use 0 to show the intent that its a dummy value?

Similarly for digest, can we use `00000000000...` to show the intent that its a dummy value?

Reply:
Some registries don't support empty, 0 size config.

The descriptor requirements describe what the value should be set for `size` and `digest`.

Since the config content is an empty json `{}`, so the size of raw content is `2` per the OCI image manifest spec. 

The digest is calculated based on the config content `{}` using the following command:

```shell
echo -n "{}" | sha256sum
```